creator-node/src/routes/nodeSync.js

```diff
@@ -308,19 +308,21 @@ async function _nodesync (req, walletPublicKeys, creatorNodeEndpoint, dbOnlySync
           transaction
         })
         const fetchedLatestBlockNumber = fetchedCNodeUser.latestBlockNumber
+        const fetchedLatestClockVal = fetchedCNodeUser.clock
 
         // Delete any previously stored data for cnodeUser in reverse table dependency order (cannot be parallelized).
         if (cnodeUser) {
           // Ensure imported data has higher blocknumber than already stored.
-          // TODO - replace this check with a clock check (!!!)
           const latestBlockNumber = cnodeUser.latestBlockNumber
+          const latestClockValue = cnodeUser.clock
 
           if (!dbOnlySync) {
```

Comment:
we can remove this if (!dbOnlySync) condition. i added it here to bypass the block number check during vector clock backfill and force secondaries to sync from the primary even if outdated because it had new vector clock information. everything should check this condition now.